### PR TITLE
Add faster site messages update

### LINKED ISSUE #3974: Addon Request: Make Message Count on Scratch Update As Fast as the Addon
### Is your request related to a problem? Please describe.
The Scratch Messages icon does not update when a message is read via the addon, and is slow to refresh when you DO get a notification.

### Describe the solution you'd like
I'm requesting an addon that updates the message amount as fast as the addon. Note that this for the Scratch WEBSITE, not for the number on the extension icon.

### Describe alternatives you've considered
Just using the addon. But it doesn't load all the time.

### Additional context
None

### PR DESCRIPTION
**Pull request by hello-smile6**
Resolves #3974

### Changes

- Create addon with name `Faster site messages badge updating` in directory `addons/faster-site-messages-update`
- Configure userscript for previously mentioned addon
- Take code from other addons until it worked

### Reason for changes

- Scratch updates the message count on webpages every 30 seconds by default. Many users would prefer the message count to be updated more often. Since the update rate used for the extension icon (2500 milliseconds) seems to be an agreed-upon interval, I used the same timing for my addon.
- Users with touchscreens may prefer to use tablet mode. On Chrome OS, the URL bar is hidden at times in tablet mode. By updating the webpage's message count faster, those users will be able to respond to messages easier.

### Tests

- Tested on various pages with clean install, including on a user profile page, a studio page, and various forum pages. All of them worked perfectly.

### Additional credits
Thank you, @webdev03 , for making sure my work made at least some sense.

### CODE REVIEW

**Review by hello-smile6 — COMMENTED** [review]
Was `console.error()` right for this?

**Comment by hello-smile6** [comment]
I'm enabling actions, since it was necessary for https://github.com/ScratchAddons/ScratchAddons/pull/3986#issuecomment-1000318215, I assume it is necessary for me as well.

**Comment by hello-smile6** [comment]
@WorldLanguages Is there anything I need to change?

**Comment by WorldLanguages** [comment]
> Since the update rate used for the extension icon (2500 milliseconds) seems to be an agreed-upon interval

This is misleading, see https://github.com/ScratchAddons/ScratchAddons/blob/master/background/handle-messages.js
`const COUNT_CHECK_INTERVAL = 30000;`

File: addons/faster-site-messages-update/userscript.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+export default async function ({ addon, console, setTimeout, setInterval, clearTimeout, clearInterval }) {
+  if (addon.tab.clientVersion === null) return;
+  if (window.location.href.includes("/projects") && window.location.href.includes("/editor")) return;
+  let msgCount;
+  // console.log("Hi from the addon!")
+  const setBadge = () => {
+    if (msgCount === null && addon.settings.get("showOffline")) {
```
Comment (RedGuy12) [inline addons/faster-site-messages-update/userscript.js:7]: What's showOffline

File: addons/faster-site-messages-update/userscript.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      // Do nothing
+    } else {
+      var targetElement;
+      if (addon.tab.clientVersion === "scratch-www") {
+        targetElement = document.querySelector(
```
Comment (RedGuy12) [inline addons/faster-site-messages-update/userscript.js:12]: Use waitForElement

File: addons/faster-site-messages-update/userscript.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        );
+      }
+      // console.log(document)
+      if (typeof msgCount == "number") {
+        targetElement.innerText = msgCount;
+      } else {
+        // Assume that message count check failed
+        console.error("Message count check failed. Result was invalid.\n\nResult was:\n".concat(msgCount));
+      }
+      targetElement = null; // Garbage collection
```
Comment (RedGuy12) [inline addons/faster-site-messages-update/userscript.js:27]: Not necessary

**Review by RedGuy12 — CHANGES_REQUESTED** [review]
How often does the badge update currently?

**Comment by Samq64** [comment]
> How often does the badge update currently?

I think it might be every 30 seconds, but I'm not sure.

**Comment by webdev03** [comment]
https://github.com/ScratchAddons/ScratchAddons/pull/4006/commits/0441650ee8c4270bb2c8087d21a8c4a06a4ef10f _interesting_ commit description

Why don't you make the amount of time a setting, and make it so that you can't spam the API by making it at least 2000ms?

**Comment by apple502j** [comment]
Don't think this is very useful. You can comment if you disagree.

**Comment by GrahamSH-LLK** [comment]
Also, this PR is abandoned.

**Pull request closed without merging**